src/lib.rs

```diff
@@ -272,6 +275,33 @@ impl StreamDeck {
         self.write_button_image(key, &image)
     }
 
+    /// Sets a button to the provided text.
+    /// Will break text over \n linebreaks
+    pub fn set_button_text(
+        &mut self,
+        key: u8,
+        colour: &Colour,
+        background: &Colour,
+        x: u32,
+        y: u32,
+        scale: Scale,
+        line_height: f32,
+        font: &Font,
+        text: &str,
+    ) -> Result<(), Error> {
+        let (width, height) = self.kind.image_size();
+        let background = Rgb([background.r, background.g, background.b]);
+        let colour = Rgb([colour.r, colour.g, colour.b]);
```

Comment:
I contemplated adding a `to_rgb()` and maybe a `to_rgba(opacity)` to Colour to clean up some of this repetition, but figured that might be best handled by a different PR.

WDYT?

Reply:
seems like a good improvement, up to you whether you included it here or in another pr, i don't particularly mind either way ^_^

Reply:
I'll do it in another PR. I'm sure I'll come back around to this at some point int the future :smile: no need to hold things up.